### ISSUE TITLE
MPD works all the time without the need

### ISSUE THREAD

**Issue by avently** [segler-alex/RadioDroid#485]
When the app goes to background MPD client tries to connect to a server. It happens in a loop. It's not needed at all while the app's UI is hidden.
You can see some logs about connections via logcat. 
So even when a player is stopped RadioDroid can't stop connecting via MPD. 
If someone wants to fix it - good, do it, please. Or I'll fix it myself someday.

**Comment by werman** [comment]
It is solved in work-in-progress branch of mine https://github.com/werman/RadioDroid/commits/feature/player-screen - by complete rewrite of mpd...

My current implementation refreshes servers only when the dialog with their selection is visible. The changes are close to be finalized and I think I'll send them soon™.

 If you could I'll appreciate testing of that branch in respect to mpd, especially its usability (from my perspective I made it more straightforward). Also better backup you data before installing my branch even if I think I didn't broke anything (I'll definitely test for this later).

**Comment by avently** [comment]
@werman 
```
FAILURE: Build failed with an exception.
* Where:
Build file '/pathtoradiodroid/app/build.gradle' line: 43
Could not get unknown property 'LastFMAPIKey'


Then, after this error was fixed:

app/src/main/java/net/programmierecke/radiodroid2/station/live/metadata/lastfm/LfmMetadataSearcher.java:13: error: self-reference in initializer
import net.programmierecke.radiodroid2.station.live.metadata.lastfm.data.LfmTrackMetadata;
```

I already forgot how I fixed the second error when I tried your build months ago.

**Comment by werman** [comment]
hmm, sorry, I'll check the last error later today.

**Comment by avently** [comment]
If a developer doesn't have lastfm api key the app shouldn't fail to compile too

**Comment by werman** [comment]
Yes, but that's something I know =) just forgot to mention.

**Comment by werman** [comment]
```
app/src/main/java/net/programmierecke/radiodroid2/station/live/metadata/lastfm/LfmMetadataSearcher.java:13: error: self-reference in initializer
import net.programmierecke.radiodroid2.station.live.metadata.lastfm.data.LfmTrackMetadata;
```

I could not reproduce this... And there is nothing suspicious.

Also I made the lastfm apikey optional.

**Comment by avently** [comment]
@werman please, make a PR and we will discuss your changes in the PR.
After your latest commit I was able to build without errors.

These are my first impressions:
1. MPD icon was checked (in "enabled state") but MPD was disabled. So the icon shows everytime that MPD is enabled. 
2. The idea to show servers with UI like you did is sooo cool. 
3. I have no idea what "play" icon does (which is located at servers card) in a dialog appeared from menu button. It just does nothing.
4. Styles are not applied to dialogs.
5. You (?) replaced buttons in stations list (links, alarm) with a text. It's ugly and should not be used. Previous implementation with icons was good.
6. "Add MPD server" button should be a "+" icon and located on the right side of a dialog.
7. "Three dots" icon is not visible on dark card (maybe because a theme is not applied to the dialog)
8. I don't think we need a dialog presented from mini player's play button. The dialog presented from MPD button is ok but just need to be fixed (play button doesn't work in it). Because I don't want to change the state of MPD server often so the dialog doesn't make sense when you already have it from the menu button. But, please, leave this dialog in situation when a user clicked on a station (usability is on a high level) 
9. Started playing a station via MPD (the station really started to play), pressed pause, pressed play again and nothing happened (no sound).
10. There is probably too much empty space below "Enable MPD", "Add MPD server" buttons.
11. Got a crash while pressing on play button on a dialog appeared from bottom mini player "play" button on a connected MPD server:
```
FATAL EXCEPTION: main
Process: net.programmierecke.radiodroid2, PID: 16314
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@8858580 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@8707b9[Shutting down, pool size = 1, active threads = 1, queued tasks = 6, completed tasks = 6]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2049)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:814)
	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:305)
	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
	at java.util.concurrent.ScheduledThreadPoolExecutor.submit(ScheduledThreadPoolExecutor.java:635)
	at net.programmierecke.radiodroid2.players.mpd.MPDClientNew.enqueueTask(MPDClientNew.java:51)
	at net.programmierecke.radiodroid2.players.mpd.MPDServersAdapter$PlayStationTask.onPostExecute(MPDServersAdapter.java:156)
	at net.programmierecke.radiodroid2.players.mpd.MPDServersAdapter$PlayStationTask.onPostExecute(MPDServersAdapter.java:106)
	at android.os.AsyncTask.finish(AsyncTask.java:667)
	at android.os.AsyncTask.-wrap1(AsyncTask.java)
	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:684)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6236)
	at java.lang.reflect.Method.invoke(Native Method)
```


Now let's talk about fullscreen player.
1. Tap on record button. Recording is showing (so cool). Tap on stop. You will not see new recording in a "Recordings" panel right after stop. Expected behaviour is to see an updated list.
2. Can you make a player higher on Z axis than the top appbar?  So the player will cover the appbar without the need to hide it with animation (it looks bad because the appbar should be underneath of the player).
3. There is a song's title in the player at the top of the fragment. And it doesn't have the same look and feel as a mini player's version (and I'm pretty sure you use white theme:) ). I suggest to you to use the same TextView as mini player does but to hide station's name (because you want to show the station's name in a different way under a song's title).
4. "Add to favorites" button can be placed at the right side of the fullscreen player (right after "next" station button).
5.  "Lyrics" and "Copy info" should be replaced with icons.
6. Station's name should be blue as the mini player's version.
7. Tags are not clickable for some reason.

Didn't see an album art since I don't have API key so can't say anything about it.

@werman you made an awesome job! I really like how player's and MPD's controls look like. The logic behind your changes makes UI simpler and more productive.

**Comment by werman** [comment]
@avently Thanks for the feedback! I'll create a wip PR in a several minutes. I'd like you to move feedback there.

**Comment by segler-alex** [comment]
is this fixed?

**Comment by avently** [comment]
Nope, on master branch problem still here. Werman said he fixed it in #495 but I don't know.

**Comment by avently** [comment]
@werman I believe the issue is fixed. Thanks!
